data/transactions/logic/eval.go

```diff
@@ -4580,64 +4612,89 @@ func (cx *EvalContext) localsReference(account stackValue, ref uint64) (basics.A
 	return addr, app, addrIdx, nil
 }
 
-func (cx *EvalContext) assetReference(ref uint64, foreign bool) (basics.AssetIndex, error) {
+func (cx *EvalContext) assetReference(ref uint64, foreign bool) (aid basics.AssetIndex, err error) {
```

Comment:
Correct. 0 is special for apps (because 0 means "current"), but not for assets, as there's no concept of the current app when running an app.

Incidentally, that means there is one more tiny difference - the ref is compare with `<=` for apps, but `<` for asas, to account for the shift.